Build fails on Archlinux

While compiling openspace-git, I am getting a build error. I have tried with a fresh build and it is still not compiling.

```
[ 85%] Building CXX object src/CMakeFiles/openspace-core.dir/scene/scene.cpp.o
[ 85%] Building CXX object src/CMakeFiles/openspace-core.dir/scene/sceneinitializer.cpp.o
[ 85%] Building CXX object src/CMakeFiles/openspace-core.dir/scene/scenegraphnode.cpp.o
[ 85%] Building CXX object src/CMakeFiles/openspace-core.dir/scene/timeframe.cpp.o
[ 85%] Building CXX object src/CMakeFiles/openspace-core.dir/scene/translation.cpp.o
In file included from /home/morphon/.cache/aurman/openspace-git/src/OpenSpace/ext/ghoul/include/ghoul/lua/lua_helper.inl:30,
                 from /home/morphon/.cache/aurman/openspace-git/src/OpenSpace/ext/ghoul/include/ghoul/lua/lua_helper.h:620,
                 from /home/morphon/.cache/aurman/openspace-git/src/OpenSpace/build/src/CMakeFiles/openspace-core.dir/cmake_pch.hxx:9,
                 from <command-line>:
/usr/include/oneapi/tbb/task_group.h:588:21: error: non-local lambda expression cannot have a capture-default
  588 |     d2::task_handle defer(F&& f) {
      |                     ^~~~~
In file included from /usr/include/oneapi/tbb/partitioner.h:47,
                 from /usr/include/oneapi/tbb/parallel_for.h:27,
                 from /usr/include/tbb/parallel_for.h:17,
                 from /usr/include/c++/14.2.1/pstl/parallel_backend_tbb.h:25,
                 from /usr/include/c++/14.2.1/pstl/parallel_backend.h:20,
                 from /usr/include/c++/14.2.1/pstl/algorithm_impl.h:22,
                 from /usr/include/c++/14.2.1/pstl/glue_execution_defs.h:50,
                 from /usr/include/c++/14.2.1/execution:39,
                 from /home/morphon/.cache/aurman/openspace-git/src/OpenSpace/src/scene/scene_lua.inl:61,
                 from /home/morphon/.cache/aurman/openspace-git/src/OpenSpace/src/scene/scene.cpp:39:
/usr/include/oneapi/tbb/task_group.h: In lambda function:
/usr/include/oneapi/tbb/task_group.h:588:26: error: expected ‘{’ before ‘(’ token
  588 |     d2::task_handle defer(F&& f) {
      |                          ^
/usr/include/oneapi/tbb/task_group.h: At global scope:
/usr/include/oneapi/tbb/task_group.h:588:28: error: expected primary-expression before ‘&&’ token
  588 |     d2::task_handle defer(F&& f) {
      |                            ^~
/usr/include/oneapi/tbb/task_group.h:588:31: error: ‘f’ was not declared in this scope
  588 |     d2::task_handle defer(F&& f) {
      |                               ^
/usr/include/oneapi/tbb/task_group.h:588:5: error: two or more data types in declaration of ‘scopeExit_588’
  588 |     d2::task_handle defer(F&& f) {
      |     ^~
[ 85%] Building CXX object src/CMakeFiles/openspace-core.dir/scripting/lualibrary.cpp.o
[ 85%] Building CXX object src/CMakeFiles/openspace-core.dir/scripting/scriptengine.cpp.o
[ 85%] Building CXX object src/CMakeFiles/openspace-core.dir/scripting/scriptscheduler.cpp.o
[ 85%] Building CXX object src/CMakeFiles/openspace-core.dir/scripting/systemcapabilitiesbinding.cpp.o
[ 85%] Building CXX object src/CMakeFiles/openspace-core.dir/util/blockplaneintersectiongeometry.cpp.o
[ 85%] Building CXX object src/CMakeFiles/openspace-core.dir/util/boxgeometry.cpp.o
[ 85%] Building CXX object src/CMakeFiles/openspace-core.dir/util/collisionhelper.cpp.o
[ 85%] Building CXX object src/CMakeFiles/openspace-core.dir/util/coordinateconversion.cpp.o
[ 85%] Building CXX object src/CMakeFiles/openspace-core.dir/util/distanceconversion.cpp.o
cc1plus: note: unrecognized command-line option ‘-Wno-unknown-attributes’ may have been intended to silence earlier diagnostics
make[2]: *** [src/CMakeFiles/openspace-core.dir/build.make:1904: src/CMakeFiles/openspace-core.dir/scene/scene.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:3939: src/CMakeFiles/openspace-core.dir/all] Error 2
make: *** [Makefile:156: all] Error 2
==> ERROR: A failure occurred in build().
    Aborting...
2025-04-26 18:36:05,540 - wrappers - makepkg - ERROR - makepkg query ['makepkg', '-cf', '--noconfirm'] failed in directory /home/morphon/.cache/aurman/openspace-git

```

Any ideas on how to fix this? I am also using GCC 14.2.1

#3615

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -466,6 +466,17 @@ target_include_directories(openspace-core
 
 target_link_libraries(openspace-core PUBLIC nlohmann_json)
 
+if (UNIX)
+  # On ArchLinux, if you have the package `oneapi` installed, this influences including
+  # the <exception> header and causes the header /usr/include/oneapi/tbb/task_group.h be
+  # pulled in, causing a compile problem
+  target_compile_definitions(
+    openspace-core
+      PUBLIC "_GLIBCXX_USE_TBB_PAR_BACKEND=0"
+      PUBLIC "_PSTL_PAR_BACKEND_SERIAL"
+  )
+endif ()
+
 target_precompile_headers(openspace-core PRIVATE
   <openspace/json.h>
   <openspace/properties/numericalproperty.h>

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -467,7 +467,7 @@ target_include_directories(openspace-core
 target_link_libraries(openspace-core PUBLIC nlohmann_json)
 
 if (UNIX)
-  # On ArchLinux, if you have the package `oneapi` installed, this influences including
+  # On Arch Linux, if you have the package `oneapi` installed, this influences including
   # the <exception> header and causes the header /usr/include/oneapi/tbb/task_group.h be
   # pulled in, causing a compile problem
   target_compile_definitions(

